emcc.py

```diff
@@ -3310,18 +3294,15 @@ def process_libraries(libs, lib_dirs, temp_files):
     logger.debug('looking for library "%s"', lib)
 
     found = False
-    for prefix in LIB_PREFIXES:
```

Comment:
the lib prefix change here is the one part I don't follow. Before we used to check both with and without `lib*`, but now we require `lib*`, is that right? If so, why is that not a possible non-NFC change?

Reply:
It seems low risk so I think it's ok as part of this PR, but I think we should remove the NFC marker and also mention the change in the changelog. Even if it's something we didn't want to support, it has been in practice so it could be a breaking change for someone. lgtm with that.

Reply:
I split out the non-NFC part to keep this change pure.  Will upload it as a separate change.